src/shippingservice/main.go

```diff
@@ -68,7 +78,12 @@ func main() {
 	if err != nil {
 		log.Fatalf("failed to listen: %v", err)
 	}
-	srv := grpc.NewServer(grpc.StatsHandler(&ocgrpc.ServerHandler{}))
+	// TOOD: replace ocgrpc with automatic OpenTelemetry grpc metrics collector
+	srv := grpc.NewServer(grpc.StatsHandler(
+		&ocgrpc.ServerHandler{}), // TODO: replace with automatic OTel grpc metrics collector when available
+		grpc.UnaryInterceptor(grpctrace.UnaryServerInterceptor(global.TraceProvider().Tracer("shipping"))),
+		grpc.StreamInterceptor(grpctrace.StreamServerInterceptor(global.TraceProvider().Tracer("shipping"))),
+	)
```

Comment:
What do the additional arguments do?

Reply:
they intercept all grpc messages to collect information, in this case the trace context so we can link calls between services